### ISSUE TITLE
Broken pubsub type when using external pubsub library

### ISSUE THREAD

**Issue by MrLuit** [libp2p/js-libp2p#1393]
- **Version**: 0.39.2

- **Platform**: N/A (all platforms)

- **Subsystem**: js-libp2p or js-libp2p-interfaces

#### Description:

As described in your example [examples/pubsub/message-filtering](https://github.com/libp2p/js-libp2p/blob/0ecc02b2a426b6dfec7b6f46d565fde41ad66954/examples/pubsub/message-filtering/README.md), when using `@chainsafe/libp2p-gossipsub`, you should be able to use `node.pubsub.topicValidators` in order to pass a function which can prevent unwanted messages from being spread. Although the `@chainsafe/libp2p-gossipsub` library does implement this feature correctly, TypeScript does not recognize this feature because your types are hardcoded instead of allowing the external library to pass their types.

This is confirmed [here](https://github.com/libp2p/js-libp2p/blob/0ecc02b2a426b6dfec7b6f46d565fde41ad66954/src/libp2p.ts#L59), where your pubsub type is assigned to the Libp2pNode class, that type definition is imported from `@libp2p/interface-pubsub` [here](https://github.com/libp2p/js-libp2p/blob/0ecc02b2a426b6dfec7b6f46d565fde41ad66954/src/libp2p.ts#L35), which is defined [here](https://github.com/libp2p/js-libp2p-interfaces/blob/399544b4e90ccc10711e86989d748b3751d52765/packages/interface-pubsub/src/index.ts#L136).

#### Steps to reproduce the error:

- Try out [your example code](https://github.com/libp2p/js-libp2p/blob/0ecc02b2a426b6dfec7b6f46d565fde41ad66954/examples/pubsub/message-filtering/1.js) for message filtering on pubsub, using TypeScript instead of JavaScript.

**Comment by mpetrunic** [comment]
@MrLuit Does this https://github.com/ChainSafe/js-libp2p-gossipsub/pull/348  resolve this issue?

**Comment by MrLuit** [comment]
No,  types are still broken for any new feature they add since gossipclient types are not passed
